plugins/tcpinfo/tcpinfo.cc

```diff
@@ -360,36 +361,37 @@ TSPluginInit(int argc, const char *argv[])
     case 'e':
       i = strtoul(optarg, &endptr, 10);
       if (*endptr != '\0' || i > 3) {
-        TSError("[tcpinfo invalid rolling-enabled argument, using default of %d, valid values are 0-3", config->rolling_enabled);
+        TSError("[tcpinfo invalid rolling-enabled argument, '%s', using default of %d, valid values are 0-3", optarg,
+                rolling_enabled);
       } else {
-        config->rolling_enabled = i;
+        rolling_enabled = i;
       }
       break;
     case 'H':
       i = strtoul(optarg, &endptr, 10);
       if (*endptr != '\0' || i > 23) {
-        TSError("[tcpinfo invalid rolling-offset-hr argument, using default of %d, valid values are 0-23",
-                config->rolling_offset_hr);
+        TSError("[tcpinfo invalid rolling-offset-hr argument, '%s', using default of %d, valid values are 0-23", optarg,
```

Comment:
All these error messages should read:
```C
TSError("[tcpinfo] invalid rolling-offset-hr argument, '%s', using default of %d", ...);
```

Note the closing ']'. There's no need to restate the valid ranges, since you documented them nicely.